service/history/historyEngine.go

```diff
@@ -972,6 +972,18 @@ Update_History_Loop:
 
 		scheduleID := token.ScheduleID
 
+		if scheduleID == common.EmptyEventID { // When called by client:CompleteActivityById
```

Comment:
This code is repeated 3 times.  Can we refactor all the activity completion methods into single one?

Reply:
Done